workload/src/Samsung.Tizen.Build.Tasks/CheckApiVersion.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Samsung.Tizen;
+
+namespace Samsung.Tizen.Build.Tasks
+{
+    public class CheckApiVersion : Task
+    {
+        [Required]
+        public string ManifestApiVersion { get; set; }
+
+        [Required]
+        public string TargetFrameWork { get; set; }
+
+        [Required]
+        public string TargetFrameworkVersion { get; set; }
+
+        [Required]
+        public string TargetPlatformVersion { get; set; }
+
+        [Required]
+        public ITaskItem[] ApiVersionList { get; set; }
+
+        string ApiVersion { get; set; }
+
+        public override bool Execute()
+        {
+            if (string.IsNullOrEmpty(ManifestApiVersion))
+            {
+                Log.LogError("ManifestApiVersion is Invalid {0}", ManifestApiVersion);
+                return false;
+            }
+
+            if (TargetFrameWork.StartsWith("tizen"))
+            {
+                //net5.0 tfm, get api version from tpv
+                if (string.IsNullOrEmpty(TargetFrameworkVersion))
+                {
+                    Log.LogError("TargetFrameworkVersion is Invalid {0}", TargetFrameworkVersion);
+                    return false;
+                }
+
+                //convert tfv to api version
+                foreach(var item in ApiVersionList)
+                {
+                    if (Regex.IsMatch(item.ItemSpec, TargetFrameworkVersion))
+                    {
+                        ApiVersion = item.GetMetadata("Value");
+                        break;
+                    }
+                }
+            }
+            else if (TargetFrameWork.StartsWith("net6.0-tizen"))
+            {
+                //net6.0 tfm, get api version from tpv
+                if (string.IsNullOrEmpty(TargetPlatformVersion))
+                {
+                    Log.LogError("TargetPlatformVersion is Invalid {0}", TargetPlatformVersion);
+                    return false;
+                }
+                ApiVersion = TargetPlatformVersion;
+            }
+
+            if (string.Compare(ManifestApiVersion, ApiVersion.ToString()) > 0)
```

Comment:
How about using `Vesion` class to store and compare version information?
https://docs.microsoft.com/en-us/dotnet/api/system.version?view=netstandard-2.0

Reply:
When tested, string converting is still required.  It would be better to leave it as a string.

Reply:
String comparing can be a problem when the api-version reaches `10`.

Reply:
Fixed.